sql/sql_table.cc

```diff
@@ -3999,21 +4017,26 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
       if (key_part_length > file->max_key_part_length() &&
           key->type != Key::FULLTEXT)
       {
-        key_part_length= file->max_key_part_length();
-	if (key->type == Key::MULTIPLE)
-	{
-	  /* not a critical problem */
-	  push_warning_printf(thd, Sql_condition::WARN_LEVEL_WARN,
-                              ER_TOO_LONG_KEY, ER_THD(thd, ER_TOO_LONG_KEY),
-                              key_part_length);
-          /* Align key length to multibyte char boundary */
-          key_part_length-= key_part_length % sql_field->charset->mbmaxlen;
-	}
-	else
-	{
-	  my_error(ER_TOO_LONG_KEY, MYF(0), key_part_length);
-	  DBUG_RETURN(TRUE);
-	}
+      /*currently Only InnoDB and MyISAM supports long unique constraints*/
```

Comment:
Again, consider a comment that does not need change when some other storage engine adds support for long unique constraints.